packages/shared/src/store/postActions.ts

```diff
@@ -166,16 +166,23 @@ export async function showPost({ post }: { post: db.Post }) {
 }
 
 export async function deletePost({ post }: { post: db.Post }) {
+  const existingPost = await db.getPost({ postId: post.id });
+
   // optimistic update
-  await db.deletePost(post.id);
+  await db.markPostAsDeleted(post.id);
+  await db.updateChannel({ id: post.channelId, lastPostId: null });
```

Comment:
trying to think if there's any weird side effects here. Guess you just lose the chatlist preview until it refetches heads or gets another new post?

Reply:
I think so, yeah.